File: leetcode3/최원준/3701. Compute Alternating Sum.py
#

'''
1. 아이디어 :


2. 시간복잡도 :
    O(n + n)

3. 자료구조/알고리즘 :


'''
class Solution:
    def alternatingSum(self, nums: List[int]) -> int:
        return sum([nums[i] if i%2==0 else -nums[i] for i in range(len(nums))])

File: leetcode3/염혜정/3701. Compute Alternating Sum.java
// 짝수는 더하고 홀수는 빼기
// o(n)

class Solution {
    public int alternatingSum(int[] nums) {
        int sum = 0;
        for (int i = 0; i<nums.length; i++) {
            if (i%2 == 0) sum += nums[i];
            else sum -= nums[i];
        }
        return sum;
    }
}
